File: aquality-appium-java-master/src/test/java/screens/SellerScreen.java
package screens;

import aquality.appium.elements.interfaces.ILabel;
import aquality.appium.screens.AndroidScreen;
import models.Seller;
import org.openqa.selenium.By;

public class SellerScreen extends AndroidScreen {

    private final ILabel lblCitySeller = getElementFactory().getLabel(By.id("tvItemSellerCity"), "Label price product");

    public SellerScreen() {
        super(By.id("lvSellerItems"), "Product screen");
    }

    private ILabel getLblNameSeller(String nameSeller) {
        return getElementFactory().getLabel(By.xpath(String.format("//android.widget.TextView[contains(@text, '%s')]", nameSeller)), "Label seller");
    }

    public Seller getSeller(String nameSeller) {
        Seller seller = new Seller();
        seller.setName(getLblNameSeller(nameSeller).getText());
        seller.setCity(lblCitySeller.getText());
        return seller;
    }
}

File: KotlinSecretCloset/src/test/kotlin/screens/SellerScreen.kt
package screens

import framework.elements.Label
import framework.screens.BaseScreen
import models.Seller
import org.openqa.selenium.By

class SellerScreen : BaseScreen(By.id("lvSellerItems"), "Product screen") {

    private val lblCitySeller: Label =
        Label(By.id("tvItemSellerCity"), "Label price product")

    private fun getLblNameSeller(nameSeller: String) =
        Label(By.xpath("//android.widget.TextView[contains(@text, '$nameSeller')]"), "Label seller")

    fun getSeller(nameSeller: String): Seller = with(Seller()) {
        name = getLblNameSeller(nameSeller).getText()
        city = lblCitySeller.getText()
        return this
    }
}